### PR TITLE
board.build

### PR DESCRIPTION
**Pull request by sojeri**
I am le ready to be le merged!

### CODE REVIEW

File: javascripts/2048.js [hunk 1]
```diff
@@ -63,8 +63,8 @@ Board.prototype.reorient = function(direction) {
 };
 
 Board.prototype.horizontalReorient = function() {
-  return this.board; // or do we want to modify the board in place?
-};
+  return this.board.slice(); // slice() will make a copy for us.
+}; // or do we want to modify the board in place?
```
Comment (corinnepingul) [inline javascripts/2048.js:67]: Hmm. This is a good question. Let's do it how you've set it up here and then just make sure we are consistent throughout. If we do it this way, we can reassign board at the end?

File: stylesheets/2048.css [hunk 2]
```diff
@@ -23,7 +23,7 @@ html {
 .tile {
   position: absolute;
   top: 0; left: 0;
-  transition: all 0.2s ease-in-out;
+  transition: all 0.1s ease-in-out; /* I think 0.1s is closer to the original game */
```
Comment (corinnepingul) [inline stylesheets/2048.css:26]: :ok_hand:

File: javascripts/2048.js [hunk 3]
```diff
@@ -155,3 +155,117 @@ Board.prototype.moveBackward = function(condensedColOrRow) {
...
+      } else { // "right" || "down"
+        // [2, 4] & we're about to unshift in this.emptyTile at eventual index 1 (4 - 2 - 1 == 1)
+        currentColumn = boardLength - array.length - 1; // currentColumn is for emptySpots positions
+        array.unshift(emptyTile);
+      }; // see ya, if
+
+      emptySpots.push([currentRow, currentColumn]); // tell emptySpots where we just filled in an empty tile
+    } // see ya, while
+    return array;
+  } // see ya, rebuild()
```
Comment (corinnepingul) [inline javascripts/2048.js:249]: What is the advantage to having this function inside the build function? Could it be extracted into it's own separate function and then called from within this function?

Comment (sojeri) [inline javascripts/2048.js:249]: it has access to the outside scope, which we need in order to push the coordinates in to the `emptySpots` array.

**Pull request merged**